### ISSUE TITLE
Kane's InvalidRequestError: Instance '<Order at 0x2b4fdc1b9890>' is not persistent within this Session

### ISSUE THREAD

**Issue by kbower** [kbower/tickettest1#1420]
_Imported from trac ticket 1420.
Created by **kbower**
Opened in trac: Jan 31, 2012, 10:35:32 AM
Last modified in trac: Jan 31, 2012, 2:40:15 PM_
***
Since upgrade from r1850 to [r2233](../commit/59fca29a0dfe54951d1f7795903beb04475b8da8)

```
[Mon Jan 30 18:36:18 2012] [error] [client 173.221.136.228] Error - <class 'sqlalchemy.exc.InvalidRequestError'>: Instance '<Order at 0x2aaaab8ccb10>' is not persistent within this Session
[Mon Jan 30 18:36:18 2012] [error] [client 173.221.136.228] URL: https://kanes.retailarchitects.com/tg/customerorder/save
[Mon Jan 30 18:36:18 2012] [error] [client 173.221.136.228] File '/home/rarch/tg2env/lib/python2.6/site-packages/WebError-0.10.1-py2.6.egg/weberror/errormiddleware.py', line 162 in __call__
[Mon Jan 30 18:36:18 2012] [error] [client 173.221.136.228]   app_iter = self.application(environ, sr_checker)
[Mon Jan 30 18:36:18 2012] [error] [client 173.221.136.228] File '/home/rarch/tg2env/lib/python2.6/site-packages/TurboGears2-2.0.1.1kb-py2.6.egg/tg/configuration.py', line 643 in wrapper
[Mon Jan 30 18:36:18 2012] [error] [client 173.221.136.228]   return app(environ, start_response)
[Mon Jan 30 18:36:18 2012] [error] [client 173.221.136.228] File '/home/rarch/tg2env/lib/python2.6/site-packages/TurboGears2-2.0.1.1kb-py2.6.egg/tg/configuration.py', line 543 in remover
[Mon Jan 30 18:36:18 2012] [error] [client 173.221.136.228]   return app(environ, start_response)
[Mon Jan 30 18:36:18 2012] [error] [client 173.221.136.228] File '/home/rarch/tg2env/lib/python2.6/site-packages/repoze.tm2-1.0a4-py2.6.egg/repoze/tm/__init__.py', line 19 in __call__
[Mon Jan 30 18:36:18 2012] [error] [client 173.221.136.228]   result = self.application(environ, save_status_and_headers)
[Mon Jan 30 18:36:18 2012] [error] [client 173.221.136.228] File '/home/rarch/tg2env/lib/python2.6/site-packages/repoze.who-1.0.10-py2.6.egg/repoze/who/middleware.py', line 107 in __call__
[Mon Jan 30 18:36:18 2012] [error] [client 173.221.136.228]   app_iter = app(environ, wrapper.wrap_start_response)
[Mon Jan 30 18:36:18 2012] [error] [client 173.221.136.228] File '/home/rarch/tg2env/lib/python2.6/site-packages/ToscaWidgets-0.9.7.1-py2.6.egg/tw/core/middleware.py', line 36 in __call__
[Mon Jan 30 18:36:18 2012] [error] [client 173.221.136.228]   return self.wsgi_app(environ, start_response)
[Mon Jan 30 18:36:18 2012] [error] [client 173.221.136.228] File '/home/rarch/tg2env/lib/python2.6/site-packages/ToscaWidgets-0.9.7.1-py2.6.egg/tw/core/middleware.py', line 59 in wsgi_app
[Mon Jan 30 18:36:18 2012] [error] [client 173.221.136.228]   resp = req.get_response(self.application)
[Mon Jan 30 18:36:18 2012] [error] [client 173.221.136.228] File 'build/bdist.linux-x86_64/egg/webob/__init__.py', line 1325 in get_response
[Mon Jan 30 18:36:18 2012] [error] [client 173.221.136.228] File 'build/bdist.linux-x86_64/egg/webob/__init__.py', line 1293 in call_application
[Mon Jan 30 18:36:18 2012] [error] [client 173.221.136.228] File '/home/rarch/tg2env/lib/python2.6/site-packages/ToscaWidgets-0.9.7.1-py2.6.egg/tw/core/resource_injector.py', line 67 in _injector
[Mon Jan 30 18:36:18 2012] [error] [client 173.221.136.228]   resp = req.get_response(app)
[Mon Jan 30 18:36:18 2012] [error] [client 173.221.136.228] File 'build/bdist.linux-x86_64/egg/webob/__init__.py', line 1325 in get_response
[Mon Jan 30 18:36:18 2012] [error] [client 173.221.136.228] File 'build/bdist.linux-x86_64/egg/webob/__init__.py', line 1293 in call_application
[Mon Jan 30 18:36:18 2012] [error] [client 173.221.136.228] File '/home/rarch/tg2env/lib/python2.6/site-packages/Beaker-1.3.1-py2.6.egg/beaker/middleware.py', line 70 in __call__
[Mon Jan 30 18:36:18 2012] [error] [client 173.221.136.228]   return self.app(environ, start_response)
[Mon Jan 30 18:36:18 2012] [error] [client 173.221.136.228] File '/home/rarch/tg2env/lib/python2.6/site-packages/Beaker-1.3.1-py2.6.egg/beaker/middleware.py', line 149 in __call__
[Mon Jan 30 18:36:18 2012] [error] [client 173.221.136.228]   return self.wrap_app(environ, session_start_response)
[Mon Jan 30 18:36:18 2012] [error] [client 173.221.136.228] File '/home/rarch/tg2env/lib/python2.6/site-packages/Routes-1.10.3-py2.6.egg/routes/middleware.py', line 130 in __call__
[Mon Jan 30 18:36:18 2012] [error] [client 173.221.136.228]   response = self.app(environ, start_response)
[Mon Jan 30 18:36:18 2012] [error] [client 173.221.136.228] File '/home/rarch/trunk/src/appserver/pylotengine/custom_middleware/tranlog.py', line 35 in __call__
[Mon Jan 30 18:36:18 2012] [error] [client 173.221.136.228]   ret = self.app(environ, start_response)
[Mon Jan 30 18:36:18 2012] [error] [client 173.221.136.228] File '/home/rarch/tg2env/lib/python2.6/site-packages/Pylons-0.9.7-py2.6.egg/pylons/wsgiapp.py', line 125 in __call__
[Mon Jan 30 18:36:18 2012] [error] [client 173.221.136.228]   response = self.dispatch(controller, environ, start_response)
[Mon Jan 30 18:36:18 2012] [error] [client 173.221.136.228] File '/home/rarch/tg2env/lib/python2.6/site-packages/Pylons-0.9.7-py2.6.egg/pylons/wsgiapp.py', line 324 in dispatch
[Mon Jan 30 18:36:18 2012] [error] [client 173.221.136.228]   return controller(environ, start_response)
[Mon Jan 30 18:36:18 2012] [error] [client 173.221.136.228] File '/home/rarch/trunk/src/appserver/pylotengine/lib/base.py', line 68 in __call__
[Mon Jan 30 18:36:18 2012] [error] [client 173.221.136.228]   return TGController.__call__(self, environ, start_response)
[Mon Jan 30 18:36:18 2012] [error] [client 173.221.136.228] File '/home/rarch/tg2env/lib/python2.6/site-packages/Pylons-0.9.7-py2.6.egg/pylons/controllers/core.py', line 221 in __call__
[Mon Jan 30 18:36:18 2012] [error] [client 173.221.136.228]   response = self._dispatch_call()
[Mon Jan 30 18:36:18 2012] [error] [client 173.221.136.228] File '/home/rarch/tg2env/lib/python2.6/site-packages/Pylons-0.9.7-py2.6.egg/pylons/controllers/core.py', line 172 in _dispatch_call
[Mon Jan 30 18:36:18 2012] [error] [client 173.221.136.228]   response = self._inspect_call(func)
[Mon Jan 30 18:36:18 2012] [error] [client 173.221.136.228] File '/home/rarch/tg2env/lib/python2.6/site-packages/Pylons-0.9.7-py2.6.egg/pylons/controllers/core.py', line 107 in _inspect_call
[Mon Jan 30 18:36:18 2012] [error] [client 173.221.136.228]   result = self._perform_call(func, args)
[Mon Jan 30 18:36:18 2012] [error] [client 173.221.136.228] File '/home/rarch/tg2env/lib/python2.6/site-packages/TurboGears2-2.0.1.1kb-py2.6.egg/tg/controllers.py', line 835 in _perform_call
[Mon Jan 30 18:36:18 2012] [error] [client 173.221.136.228]   self, controller, params, remainder=remainder)
[Mon Jan 30 18:36:18 2012] [error] [client 173.221.136.228] File '/home/rarch/tg2env/lib/python2.6/site-packages/TurboGears2-2.0.1.1kb-py2.6.egg/tg/controllers.py', line 172 in _perform_call
[Mon Jan 30 18:36:18 2012] [error] [client 173.221.136.228]   output = controller(*remainder, **dict(params))
[Mon Jan 30 18:36:18 2012] [error] [client 173.221.136.228] File '/home/rarch/trunk/src/appserver/pylotengine/controllers/__init__.py', line 15 in expose_ie_json_decorator
[Mon Jan 30 18:36:18 2012] [error] [client 173.221.136.228]   retval = func(*args, **params)
[Mon Jan 30 18:36:18 2012] [error] [client 173.221.136.228] File '/home/rarch/trunk/src/appserver/pylotengine/controllers/__init__.py', line 51 in decorated_func
[Mon Jan 30 18:36:18 2012] [error] [client 173.221.136.228]   return func(*args, **params)
[Mon Jan 30 18:36:18 2012] [error] [client 173.221.136.228] File '/home/rarch/trunk/src/appserver/pylotengine/controllers/__init__.py', line 65 in decorated_func
[Mon Jan 30 18:36:18 2012] [error] [client 173.221.136.228]   return func(*args, **params)
[Mon Jan 30 18:36:18 2012] [error] [client 173.221.136.228] File '/home/rarch/trunk/src/appserver/pylotengine/controllers/customerorder.py', line 24 in save
[Mon Jan 30 18:36:18 2012] [error] [client 173.221.136.228]   response = self._save(*args, **params)
[Mon Jan 30 18:36:18 2012] [error] [client 173.221.136.228] File '/home/rarch/trunk/src/appserver/pylotengine/controllers/__init__.py', line 36 in modify_database_environ
[Mon Jan 30 18:36:18 2012] [error] [client 173.221.136.228]   return func(*args, **params)
[Mon Jan 30 18:36:18 2012] [error] [client 173.221.136.228] File '/home/rarch/trunk/src/appserver/pylotengine/controllers/dbentity.py', line 343 in _save
[Mon Jan 30 18:36:18 2012] [error] [client 173.221.136.228]   return self._save_obj(obj, **kwargs)
[Mon Jan 30 18:36:18 2012] [error] [client 173.221.136.228] File '/home/rarch/trunk/src/appserver/pylotengine/controllers/dbentity.py', line 350 in _save_obj
[Mon Jan 30 18:36:18 2012] [error] [client 173.221.136.228]   merged_obj = self.modelClass.save(passed_obj, controller_validate)
[Mon Jan 30 18:36:18 2012] [error] [client 173.221.136.228] File '/home/rarch/trunk/src/appserver/pylotengine/model/objects.py', line 507 in save
[Mon Jan 30 18:36:18 2012] [error] [client 173.221.136.228]   obj._system_updates()
[Mon Jan 30 18:36:18 2012] [error] [client 173.221.136.228] File '/home/rarch/trunk/src/appserver/pylotengine/model/objects.py', line 10051 in _system_updates
[Mon Jan 30 18:36:18 2012] [error] [client 173.221.136.228]   o._process_payments()
[Mon Jan 30 18:36:18 2012] [error] [client 173.221.136.228] File '/home/rarch/trunk/src/appserver/pylotengine/model/objects.py', line 4114 in _process_payments
[Mon Jan 30 18:36:18 2012] [error] [client 173.221.136.228]   for affected_order in ArTran.record_payment(passed_payment, ArTran.DEPOSIT_IF_NEWORDER, self, self.customerorder.orders):
[Mon Jan 30 18:36:18 2012] [error] [client 173.221.136.228] File '/home/rarch/trunk/src/appserver/pylotengine/model/objects.py', line 7124 in record_payment
[Mon Jan 30 18:36:18 2012] [error] [client 173.221.136.228]   record(payment)
[Mon Jan 30 18:36:18 2012] [error] [client 173.221.136.228] File '/home/rarch/trunk/src/appserver/pylotengine/model/objects.py', line 7056 in record
[Mon Jan 30 18:36:18 2012] [error] [client 173.221.136.228]   payment._record_payment(codeid, order)
[Mon Jan 30 18:36:18 2012] [error] [client 173.221.136.228] File '/home/rarch/trunk/src/appserver/pylotengine/model/objects.py', line 7182 in _record_payment
[Mon Jan 30 18:36:18 2012] [error] [client 173.221.136.228]   DBSession.expire(order, ['transactions'])
[Mon Jan 30 18:36:18 2012] [error] [client 173.221.136.228] File 'build/bdist.linux-x86_64/egg/sqlalchemy/orm/scoping.py', line 113 in do
[Mon Jan 30 18:36:18 2012] [error] [client 173.221.136.228] File 'build/bdist.linux-x86_64/egg/sqlalchemy/orm/session.py', line 1079 in expire
[Mon Jan 30 18:36:18 2012] [error] [client 173.221.136.228]   self._expire_state(state, attribute_names)
[Mon Jan 30 18:36:18 2012] [error] [client 173.221.136.228] File 'build/bdist.linux-x86_64/egg/sqlalchemy/orm/session.py', line 1082 in _expire_state
[Mon Jan 30 18:36:18 2012] [error] [client 173.221.136.228]   self._validate_persistent(state)
[Mon Jan 30 18:36:18 2012] [error] [client 173.221.136.228] File 'build/bdist.linux-x86_64/egg/sqlalchemy/orm/session.py', line 1418 in _validate_persistent
[Mon Jan 30 18:36:18 2012] [error] [client 173.221.136.228]   mapperutil.state_str(state))
[Mon Jan 30 18:36:18 2012] [error] [client 173.221.136.228] InvalidRequestError: Instance '<Order at 0x2aaaab8ccb10>' is not persistent within this Session
[Mon Jan 30 18:36:18 2012] [error] [client 173.221.136.228]
[Mon Jan 30 18:36:18 2012] [error] [client 173.221.136.228]
[Mon Jan 30 18:36:18 2012] [error] [client 173.221.136.228] CGI Variables
[Mon Jan 30 18:36:18 2012] [error] [client 173.221.136.228] -------------
[Mon Jan 30 18:36:18 2012] [error] [client 173.221.136.228]   CONTENT_LENGTH: '-1'
[Mon Jan 30 18:36:18 2012] [error] [client 173.221.136.228]   CONTENT_TYPE: 'application/x-www-form-urlencoded'
[Mon Jan 30 18:36:18 2012] [error] [client 173.221.136.228]   DOCUMENT_ROOT: '/home/rarch/trunk/src/appserver/wsgi-config/flex'
[Mon Jan 30 18:36:18 2012] [error] [client 173.221.136.228]   GATEWAY_INTERFACE: 'CGI/1.1'
[Mon Jan 30 18:36:18 2012] [error] [client 173.221.136.228]   HTTPS: '1'
[Mon Jan 30 18:36:18 2012] [error] [client 173.221.136.228]   HTTP_ACCEPT: '*/*'
[Mon Jan 30 18:36:18 2012] [error] [client 173.221.136.228]   HTTP_ACCEPT_ENCODING: 'gzip, deflate'
[Mon Jan 30 18:36:18 2012] [error] [client 173.221.136.228]   HTTP_ACCEPT_LANGUAGE: 'en-US'
[Mon Jan 30 18:36:18 2012] [error] [client 173.221.136.228]   HTTP_AUTHORIZATION: 'Basic U01BTlNPOnZ2ZXZ4MjI='
[Mon Jan 30 18:36:18 2012] [error] [client 173.221.136.228]   HTTP_CACHE_CONTROL: 'no-cache'
[Mon Jan 30 18:36:18 2012] [error] [client 173.221.136.228]   HTTP_CONNECTION: 'Keep-Alive'
[Mon Jan 30 18:36:18 2012] [error] [client 173.221.136.228]   HTTP_COOKIE: 'pylotengine=fc90b5b4a7ea2ee36160c2bd350360c06cd6f1eeea9f54c50b134771976ad79ca3d2dc40'
[Mon Jan 30 18:36:18 2012] [error] [client 173.221.136.228]   HTTP_HOST: 'kanes.retailarchitects.com'
[Mon Jan 30 18:36:18 2012] [error] [client 173.221.136.228]   HTTP_REFERER: 'https://kanes.retailarchitects.com/SalesPylot.swf/[[DYNAMIC]]/6'
[Mon Jan 30 18:36:18 2012] [error] [client 173.221.136.228]   HTTP_USER_AGENT: 'Mozilla/4.0 (compatible; MSIE 8.0; Windows NT 5.1; Trident/4.0; .NET CLR 1.1.4322; .NET CLR 2.0.50727; .NET CLR 3.0.04506.30; .NET CLR 3.0.04506.648; .NET CLR 3.5.21022)'
[Mon Jan 30 18:36:18 2012] [error] [client 173.221.136.228]   HTTP_X_FLASH_VERSION: '10,1,102,64'
[Mon Jan 30 18:36:18 2012] [error] [client 173.221.136.228]   PATH_INFO: '/customerorder/save'
[Mon Jan 30 18:36:18 2012] [error] [client 173.221.136.228]   PATH_TRANSLATED: '/home/rarch/trunk/src/appserver/wsgi-config/flex/customerorder/save'
[Mon Jan 30 18:36:18 2012] [error] [client 173.221.136.228]   REMOTE_ADDR: '173.221.136.228'
[Mon Jan 30 18:36:18 2012] [error] [client 173.221.136.228]   REMOTE_PORT: '53797'
[Mon Jan 30 18:36:18 2012] [error] [client 173.221.136.228]   REMOTE_USER: 'SMANSO'
[Mon Jan 30 18:36:18 2012] [error] [client 173.221.136.228]   REQUEST_METHOD: 'POST'
[Mon Jan 30 18:36:18 2012] [error] [client 173.221.136.228]   REQUEST_URI: '/tg/customerorder/save'
[Mon Jan 30 18:36:18 2012] [error] [client 173.221.136.228]   SCRIPT_FILENAME: '/home/rarch/trunk/src/appserver/wsgi-config/wsgi-deployment.py'
[Mon Jan 30 18:36:18 2012] [error] [client 173.221.136.228]   SCRIPT_NAME: '/tg'
[Mon Jan 30 18:36:18 2012] [error] [client 173.221.136.228]   SERVER_ADDR: '173.204.243.250'
[Mon Jan 30 18:36:18 2012] [error] [client 173.221.136.228]   SERVER_ADMIN: 'root@localhost'
[Mon Jan 30 18:36:18 2012] [error] [client 173.221.136.228]   SERVER_NAME: 'kanes.retailarchitects.com'
[Mon Jan 30 18:36:18 2012] [error] [client 173.221.136.228]   SERVER_PORT: '443'
[Mon Jan 30 18:36:18 2012] [error] [client 173.221.136.228]   SERVER_PROTOCOL: 'HTTP/1.1'
[Mon Jan 30 18:36:18 2012] [error] [client 173.221.136.228]   SERVER_SIGNATURE: '<address>Apache/2.2.3 (CentOS) Server at kanes.retailarchitects.com Port 443</address>\\n'
[Mon Jan 30 18:36:18 2012] [error] [client 173.221.136.228]   SERVER_SOFTWARE: 'Apache/2.2.3 (CentOS)'
[Mon Jan 30 18:36:18 2012] [error] [client 173.221.136.228]
[Mon Jan 30 18:36:18 2012] [error] [client 173.221.136.228]
[Mon Jan 30 18:36:18 2012] [error] [client 173.221.136.228] WSGI Variables
[Mon Jan 30 18:36:18 2012] [error] [client 173.221.136.228] --------------
[Mon Jan 30 18:36:18 2012] [error] [client 173.221.136.228]   application: <function wrapper at 0x2b4fdba1b230>
[Mon Jan 30 18:36:18 2012] [error] [client 173.221.136.228]   beaker.cache: <beaker.cache.CacheManager object at 0x2b4fdb96d450>
[Mon Jan 30 18:36:18 2012] [error] [client 173.221.136.228]   beaker.get_session: <bound method SessionMiddleware._get_session of <beaker.middleware.SessionMiddleware object at 0x2b4fdb96d290>>
[Mon Jan 30 18:36:18 2012] [error] [client 173.221.136.228]   beaker.session: {'_accessed_time': 1327966573.085501}
[Mon Jan 30 18:36:18 2012] [error] [client 173.221.136.228]   downgrade-1.0: '1'
[Mon Jan 30 18:36:18 2012] [error] [client 173.221.136.228]   force-response-1.0: '1'
[Mon Jan 30 18:36:18 2012] [error] [client 173.221.136.228]   mod_wsgi.application_group: ''
[Mon Jan 30 18:36:18 2012] [error] [client 173.221.136.228]   mod_wsgi.callable_object: 'application'
[Mon Jan 30 18:36:18 2012] [error] [client 173.221.136.228]   mod_wsgi.handler_script: ''
[Mon Jan 30 18:36:18 2012] [error] [client 173.221.136.228]   mod_wsgi.input_chunked: '0'
[Mon Jan 30 18:36:18 2012] [error] [client 173.221.136.228]   mod_wsgi.listener_host: ''
[Mon Jan 30 18:36:18 2012] [error] [client 173.221.136.228]   mod_wsgi.listener_port: '443'
[Mon Jan 30 18:36:18 2012] [error] [client 173.221.136.228]   mod_wsgi.process_group: 'rarch'
[Mon Jan 30 18:36:18 2012] [error] [client 173.221.136.228]   mod_wsgi.request_handler: 'wsgi-script'
[Mon Jan 30 18:36:18 2012] [error] [client 173.221.136.228]   mod_wsgi.script_reloading: '1'
[Mon Jan 30 18:36:18 2012] [error] [client 173.221.136.228]   mod_wsgi.version: (3, 3)
[Mon Jan 30 18:36:18 2012] [error] [client 173.221.136.228]   nokeepalive: '1'
[Mon Jan 30 18:36:18 2012] [error] [client 173.221.136.228]   not_serialized_properties: {<class 'pylotengine.model.objects.OrderDetail'>: ['movedfromorderuid', 'salesperson1', 'salesperson2', 'salesperson1', 'salesperson2'], <class 'pylotengine.model.objects.Order'>: ['salesperson1', 'salesperson2']}
[Mon Jan 30 18:36:18 2012] [error] [client 173.221.136.228]   paste.registry: <paste.registry.Registry object at 0x2aaab0920e10>
[Mon Jan 30 18:36:18 2012] [error] [client 173.221.136.228]   paste.throw_errors: True
[Mon Jan 30 18:36:18 2012] [error] [client 173.221.136.228]   pylons.action_method: <bound method RootController.routes_placeholder of <pylotengine.controllers.root.RootController object at 0x2aaaab8cce90>>
[Mon Jan 30 18:36:18 2012] [error] [client 173.221.136.228]   pylons.controller: <pylotengine.controllers.root.RootController object at 0x2aaaab8cce90>
[Mon Jan 30 18:36:18 2012] [error] [client 173.221.136.228]   pylons.environ_config: {'session': 'beaker.session', 'cache': 'beaker.cache'}
[Mon Jan 30 18:36:18 2012] [error] [client 173.221.136.228]   pylons.pylons: <pylons.util.PylonsContext object at 0x2aaaab8cc390>
[Mon Jan 30 18:36:18 2012] [error] [client 173.221.136.228]   pylons.routes_dict: {'url': u'customerorder/save', 'action': u'routes_placeholder', 'controller': u'root'}
[Mon Jan 30 18:36:18 2012] [error] [client 173.221.136.228]   pylons.status_code_redirect: None
[Mon Jan 30 18:36:18 2012] [error] [client 173.221.136.228]   pylotengine.flex_detected: True
[Mon Jan 30 18:36:18 2012] [error] [client 173.221.136.228]   pylotengine.flex_revision: '2233'
[Mon Jan 30 18:36:18 2012] [error] [client 173.221.136.228]   repoze.tm.active: True
[Mon Jan 30 18:36:18 2012] [error] [client 173.221.136.228]   repoze.what.adapters: {'groups': {'sql_auth': <repoze.what.plugins.sql.adapters.SqlGroupsAdapter object at 0x2b4fdb96d8d0>}, 'permissions': {'sql_auth': <repoze.what.plugins.sql.adapters.SqlPermissionsAdapter object at 0x2b4fdb96d950>}}
[Mon Jan 30 18:36:18 2012] [error] [client 173.221.136.228]   repoze.what.credentials: {'repoze.what.userid': 'SMANSO', 'groups': (u'sales',), 'permissions': (u'ordernew', u'productsalesinv', u'moduleproducts', u'ordercreatesale', u'ordertypesale', u'ordercashier', u'orderquery', u'kanprintinvoiceanyway', u'ordereditownlimited', u'moduleorders', u'orderpay', u'orderedit', u'modulecustomers')}
[Mon Jan 30 18:36:18 2012] [error] [client 173.221.136.228]   repoze.who.identity: <repoze.who identity (hidden, dict-like) at 47621998448976>
[Mon Jan 30 18:36:18 2012] [error] [client 173.221.136.228]   repoze.who.logger: <logging.Logger instance at 0x2b4fd14a6950>
[Mon Jan 30 18:36:18 2012] [error] [client 173.221.136.228]   repoze.who.plugins: {'form': <FriendlyFormPlugin 47621986507280>, 'main_identifier': <FriendlyFormPlugin 47621986507280>, 'authorization_md': <repoze.what.middleware.AuthorizationMetadata object at 0x2b4fdb96da90>, 'api': <BasicAuthPlugin 47621926194448>, 'cookie': <AuthTktCookiePlugin 47621986507216>, 'sqlauth': <repoze.who.plugins.sa.SQLAlchemyAuthenticatorPlugin object at 0x2b4fdb96d990>, 'sql_user_md': <repoze.who.plugins.sa.SQLAlchemyUserMDPlugin object at 0x2b4fdb96da50>}
[Mon Jan 30 18:36:18 2012] [error] [client 173.221.136.228]   requestsiteid: '10'
[Mon Jan 30 18:36:18 2012] [error] [client 173.221.136.228]   routes.route: <routes.route.Route object at 0x2b4fd8027290>
[Mon Jan 30 18:36:18 2012] [error] [client 173.221.136.228]   routes.url: <routes.util.URLGenerator object at 0x2aaab1953390>
[Mon Jan 30 18:36:18 2012] [error] [client 173.221.136.228]   serialized_properties: {<class 'pylotengine.model.objects.CustomerOrder'>: [('currentorderid', True)]}
[Mon Jan 30 18:36:18 2012] [error] [client 173.221.136.228]   ssl-unclean-shutdown: '1'
[Mon Jan 30 18:36:18 2012] [error] [client 173.221.136.228]   toscawidgets.framework: <tw.mods.base.HostFramework object at 0x2b4fdb96d410>
[Mon Jan 30 18:36:18 2012] [error] [client 173.221.136.228]   toscawidgets.prefix: '/toscawidgets'
[Mon Jan 30 18:36:18 2012] [error] [client 173.221.136.228]   webob._parsed_post_vars: (MultiDict([('CustomerOrder', '{"restricted_attributes":null,"customer":{"restricted_attributes":null,"firstname":"BRYAN","companyname":null,"customerid":"49807","middlename":null,"phonenumber":"813-944-9713","businessphone":"813-944-9713","type":"I","address1":"13101 Shadow Run Blvd","address2":null,"city":"Riverview","zipcode":"33569","countryid":null,"email":"BS2trains@aol.com","accountopendate":"2011-11-01 00:00:00","taxexemptionid":null,"recordgrouping":null,"noemail":null,"taxexemptionnumber":null,"fullname":"BRYAN SIMKANICH","creatingCustomer":false,"picture_url":"resources/images/blankuser.jpg","state":"FL","warnings":[],"lastname":"SIMKANICH","extension":null},"orders":[{"restricted_attributes":null,"orderbalance":null,"transactions":null,"balance":null,"deliverydate":"2012-02-10 00:00:00","fromsiteid":"17","deliverytype":"D","orderid":null,"ordersiteid":"10","mainorderid":null,"type":"SAL","orderdate":"2012-01-30 00:00:00","orderlines":[{"restricted_attrib...t([('Cu...}')])>)
[Mon Jan 30 18:36:18 2012] [error] [client 173.221.136.228]   webob._parsed_query_vars: (MultiDict([]), '')
[Mon Jan 30 18:36:18 2012] [error] [client 173.221.136.228]   webob.adhoc_attrs: {'errors': 'ignore', 'language': 'en-us', 'start_response': <function repl_start_response at 0x2aaab077cc08>, 'response_type': None, 'response_ext': None, 'identity': <repoze.who identity (hidden, dict-like) at 47621998448976>}
[Mon Jan 30 18:36:18 2012] [error] [client 173.221.136.228]   wsgi process: 'Multi process AND threads (?)'
[Mon Jan 30 18:36:18 2012] [error] [client 173.221.136.228]   wsgi.file_wrapper: <built-in method file_wrapper of mod_wsgi.Adapter object at 0x2aaab034d828>
[Mon Jan 30 18:36:18 2012] [error] [client 173.221.136.228]   wsgi.version: (1, 1)
[Mon Jan 30 18:36:18 2012] [error] [client 173.221.136.228]   wsgiorg.routing_args: (<routes.util.URLGenerator object at 0x2aaab1953390>, {'url': u'customerorder/save', 'action': u'routes_placeholder', 'controller': u'root'})
[Mon Jan 30 18:36:18 2012] [error] [client 173.221.136.228]   <symbol 'FOR_CLIENT_WRITE>: True
[Mon Jan 30 18:36:18 2012] [error] [client 173.221.136.228]   <symbol 'SAVING>: True
[Mon Jan 30 18:36:18 2012] [error] [client 173.221.136.228]   <symbol 'MODIFIES_DATABASE>: True
[Mon Jan 30 18:36:18 2012] [error] [client 173.221.136.228] ------------------------------------------------------------
```

**Comment by kbower** [comment]
_**kbower** trac comment on Jan 31, 2012, 2:40:15 PM:_

(Changeset [[2249]](../commit/c5891a830eb31fb807ce5244c1849da54eaa6fcb))
 * sqlalchemy upgrade from [[2113]](../commit/3aa25522a17fffb5497ec65c0b1e6a7c0eb8f8f7) honors expunge cascade, which causes problems.  Remove this by default from relationships and backrefs
   * fixes #1415 (System Rule 'Order.process_calculateddeliverycharge' caused the following exception:AttributeError("'Order' object has no attribute 'passed_obj')
   * presumably also fixes #1420 (InvalidRequestError: Instance '<Order at 0x2b4fdc1b9890>' is not persistent within this Session)

**Comment by kbower** [comment]
_**kbower** trac status change on Jan 31, 2012, 2:40:15 PM:_ **closed**
